source/common/config/grpc_mux_impl.cc

```diff
@@ -336,6 +337,15 @@ void GrpcMuxImpl::expiryCallback(absl::string_view type_url,
 
     watch->callbacks_.onConfigUpdate({}, found_resources_for_watch, "");
   }
+
+  // Now that we've forgotten about the resource, enqueue a new request with an empty version to
+  // signal to the management server that we need an update. The check here ensures that we only
+  // send schedule one request, even if multiple resources for the same type expires around the same
+  // time.
+  if (!api_state->request_.version_info().empty()) {
```

Comment:
Would it be possible not to rely on an implicit state here? Perhaps adding another state field to ApiState would work? Perhaps something like `has_expired_resources`?

Reply:
I don't have a strong opinion here, to me conceptually this is issuing a new request because we're changing the version, so the version is the state we're tracking. 

I'm not even sure if we should be issuing a new request here, I don't think delta does. It might be sufficient to just reset the the version string, as the main problem with the current TTL behavior is around losing control plane connectivity and we'd issue a new request upon reconnecting. @markdroth @htuch for xDS thoughts

Reply:
As per discussion on slack, I don't think this is the right approach.  I think the xDS server should proactively resend the resources in this case, because basically the only cases where it can safely avoid resending are the LDS and CDS wildcard cases.